sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -87,6 +87,7 @@ class Analyzer(
       EliminateUnions),
     Batch("Resolution", fixedPoint,
       ResolveRelations ::
+      SubstituteHints ::
```

Comment:
In each batch, the order of rules should not matter. That means, the rule `SubstituteHints` impacts the other rules. Please fix it.